### PR TITLE
Add permanent item card

### PR DESCRIPTION
**Pull request by NastyaKolomoets**

### CODE REVIEW

File: src/app/cards/munchkin/cards/treasures/permanent-item/model/permanent-item-card.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+
+import { PermanentItemType } from './permanent-item-type';
+import { ImageBottomLayout } from 'src/app/cards/infrastructure/cards';
+import { TreasureCard } from '../../treasure-card';
+
+/**
+ * Represents a permanent item card. Extends [[DoorCard]].
```
Comment (welcomeherenowleave) [inline src/app/cards/munchkin/cards/treasures/permanent-item/model/permanent-item-card.ts:7]: `[[DoorCard]]` : )

**Review by welcomeherenowleave — APPROVED** [review]
Approved with a minor comment.

File: src/app/cards/munchkin/cards/treasures/permanent-item/card/permanent-item-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Component, Input } from '@angular/core';
+import { CardTemplate } from 'src/app/cards/infrastructure/cards';
+import { PermanentItemCard } from '../model/permanent-item-card';
+
+@Component({
+  selector: 'app-permanent-item-card',
+  templateUrl: './permanent-item-card.component.html'
+})
+export class PermanentItemCardComponent implements CardTemplate {
```
Comment (YaroslavKormushyn) [inline src/app/cards/munchkin/cards/treasures/permanent-item/card/permanent-item-card.component.ts:9]: Use `ReusableItem` as suggested [here](https://github.com/NastyaKolomoets/card-editor/pull/46#discussion_r484066705).

File: src/app/cards/munchkin/cards/treasures/permanent-item/edit/permanent-item-edit.component.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  <app-editable-string [(string)]="card.bonus" [label]="'bonus'">
+  </app-editable-string>
+  <app-editable-string [(string)]="card.specificity" [label]="'specificity'">
+  </app-editable-string>
+  <app-editable-rich-text [(text)]="card.description" [label]="'description'" [multiline]="'true'">
+  </app-editable-rich-text>
+  <app-editable-string [(string)]="card.kind" [label]="'kind'">
+  </app-editable-string>
+  <input type="checkbox" [(ngModel)]="card.isHuge" [value]="card.isHuge"> Huge
+  <app-editable-number [(number)]="card.price" min="0" max="2000" step="100" [label]="'price'">
```
Comment (YaroslavKormushyn) [inline src/app/cards/munchkin/cards/treasures/permanent-item/edit/permanent-item-edit.component.html:13]: How about having `min` and `max` set in the environment or per-implementation basis?
Also, `step` could be moved out to a component constant.

File: src/app/cards/munchkin/cards/treasures/permanent-item/model/permanent-item-card.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  kind: string;
+  isHuge: boolean;
+
+  constructor(o: any = null) {
+    super(PermanentItemType, o);
+    this.bonus = o?.bonus || '1';
+    this.specificity = o?.specificity || '';
+    this.kind = o?.kind || '';
+    this.isHuge = o?.isHuge || false;
+    this.layout = o?.layout || ImageBottomLayout;
```
Comment (YaroslavKormushyn) [inline src/app/cards/munchkin/cards/treasures/permanent-item/model/permanent-item-card.ts:22]: We could possibly inject a default configuration on a per-card type basis, so we don't have to change these in each component separately.

**Review by YaroslavKormushyn — APPROVED** [review]
Approved with suggestions.

File: src/app/cards/munchkin/cards/treasures/permanent-item/edit/permanent-item-edit.component.html (hunk 3 above)
Comment (NastyaKolomoets) [reply src/app/cards/munchkin/cards/treasures/permanent-item/edit/permanent-item-edit.component.html]: Good point, created ticket for that

**Pull request merged**